src/hooks/useActiveNetwork.ts

```diff
@@ -86,12 +91,20 @@ export function useActiveNetwork() {
               }
               successCallback && successCallback()
             } catch (addError) {
-              console.error(addError)
+              console.error('add', addError)
+              // This is a workaround solution for Brave
+              if (addError?.message?.includes('rejected the request.')) {
```

Comment:
we have `checkForBraveBrowser` function.  you can use it to make sure only Brave wallet will be applied this codes

Reply:
It's not like that. For Metamask the message is 'User rejected the request.' but for Brave it is 'The user rejected the request.' So I check if the message includes 'rejected the request.' string I will show the error popup. Maybe my comment made you confused.

Reply:
I mean `checkForBraveBrowser () && addError?.message?.includes('rejected the request.')`

Reply:
No need checkForBraveBrowser function. This code work and should be applied for both Metamask and Brave

Reply:
ah, ok bro

Reply:
anyway why dont we check error code. I remember that code is -4001.

Reply:
Because i remember 4001 can happened in many other cases like deny a transaction, deny a stake, ...etc.. so I do that. but you are right. This one is try-catch a switch/add chain to wallet so it should be fine if we check 4001 error code.

Reply:
I think, it's better to write as `addError?.message === 'User rejected the request.' || addError?.message === 'The user rejected the request.'` than write as `.includes`. Also add some comment to say which is metamask, which is brave. Let the code declarative itself.

Reply:
@namgold I just change to check error code 4001 as Danh's suggestion. it is much more clean I think